contracts/BatchExchange.sol

```diff
@@ -572,7 +577,19 @@ contract BatchExchange is EpochTokenLocker {
       * @param executedAmount proportion of order's requested sellAmount that was filled.
       */
     function revertRemainingOrder(address owner, uint16 orderId, uint128 executedAmount) private {
-        orders[owner][orderId].usedAmount = orders[owner][orderId].usedAmount.sub(executedAmount).toUint128();
+        if (isOrderTrackingItsUsedAmount(orders[owner][orderId])) {
+            orders[owner][orderId].usedAmount = orders[owner][orderId].usedAmount.sub(executedAmount).toUint128();
+        }
+    }
+
+    /** @dev Checks whether an order is intended to track its usedAmount
+      * @param order order under inspection
+      * @return true if the given order does track its usedAmount
+      */
+    function isOrderTrackingItsUsedAmount(Order memory order) private pure returns (bool) {
```

Comment:
naming nit: maybe invert the logic and call it `isOrderWithUnlimitedAmount`?

Reply:
or simply `isOrderUnlimited`